frontend/src/metabase/redux/settings.ts

```diff
@@ -35,13 +35,18 @@ export const settings = createReducer(
   },
 );
 
+type RequiredUserSettings = {
+  [K in keyof UserSettings]-?: UserSettings[K];
+};
```

Comment:
There is a utility type for that

```suggestion
type RequiredUserSettings = Required<UserSettings>;
```

Reply:
TIL, thanks.